### PR TITLE
Update dependency go modules for k8s v1.28.0

### PR DESCRIPTION
**Pull request by Sneha-at**
Ran kubernetes-csi/csi-release-tools go-get-kubernetes.sh -p 1.28.0
            
```release-note
Update kubernetes dependencies to v1.28.0
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -67,3 +67,17 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
```
Comment (Sneha-at) [inline go.mod:69]: @msau42 Should I delete these replace statements here?

File: go.mod [hunk 2]
```diff
@@ -6,17 +6,17 @@ require (
 	github.com/container-storage-interface/spec v1.8.0
 	github.com/google/gofuzz v1.2.0 // indirect
 	github.com/imdario/mergo v0.3.12 // indirect
-	github.com/kubernetes-csi/csi-lib-utils v0.13.0
+	github.com/kubernetes-csi/csi-lib-utils v0.15.0
```
Comment (msau42) [inline go.mod:9]: Keep this at 0.14.0 until we resolve the dependency issue.

File: go.mod (hunk 1 above)
Comment (msau42) [reply go.mod:74]: If the script is adding it then let's keep it. We should investigate how to modify the script to remove it if it's unnecessary.

File: go.mod (hunk 2 above)
Comment (Sneha-at) [reply go.mod:9]: This is causing dependency issue only for certain libraries from the otel package. It did not throw any error for "external-resizer" in `make test`. SHould we still use `v14` (may be to keep all sidecars on same dependency?) then we might want to as well revisit [node-driver-registrar](https://github.com/kubernetes-csi/node-driver-registrar/pull/323/files)

Comment (msau42) [reply go.mod:9]: I think it's better to be consistent across all the sidecars.

Comment (Sneha-at) [reply go.mod]: Updated `csi-lib-utils` to `v0.14.0`

**Pull request merged**